salt/cloud/clouds/saltify.py

```diff
@@ -210,12 +216,61 @@ def show_instance(name, call=None):
 
 def create(vm_):
     '''
-    Provision a single machine
+    if configuration parameter ``deploy`` is ``True``,
+
+        Provision a single machine, adding its keys to the salt master
+
+    else,
+
+        Test ssh connections to the machine
+
+    Configuration parameters:
+
+    - deploy:  (see above)
+    - provider:  name of entry in ``salt/cloud.providers.d/???`` file
+    - ssh_host: IP address or DNS name of the new machine
+    - ssh_username:  name used to log in to the new machine
+    - ssh_password:  password to log in (unless key_filename is used)
+    - key_filename:  (optional) SSH private key for passwordless login
+    - ssh_port: (default=22) TCP port for SSH connection
+    - wake_on_lan_mac:  (optional) hardware (MAC) address for wake on lan
+    - wol_sender_node:  (optional) salt minion to send wake on lan command
+    - wol_boot_wait:  (default=30) seconds to delay while client boots
+    - force_minion_config: (optional) replace the minion configuration files on the new machine
+
+    See also
+    :ref:`Miscellaneous Salt Cloud Options <misc-salt-cloud-options>`
+    and
+    :ref:`Getting Started with Saltify <getting-started-with-saltify>`
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt-cloud -p mymachine my_new_id
     '''
     deploy_config = config.get_cloud_config_value(
         'deploy', vm_, __opts__, default=False)
 
     if deploy_config:
+        wol_mac = config.get_cloud_config_value(
+            'wake_on_lan_mac', vm_, __opts__, default='')
+        wol_host = config.get_cloud_config_value(
+            'wol_sender_node', vm_, __opts__, default='')
+        if wol_mac and wol_host:
+            log.info('sending wake-on-lan to %s using node %s',
+                     wol_mac, wol_host)
+            local = salt.client.LocalClient()
+            if isinstance(wol_mac, six.string_types):
+                wol_mac = [wol_mac]  # a smart user may have passed more params
+            ret = local.cmd(wol_host, 'network.wol', wol_mac)
```

Comment:
why not just use the wol runner?

https://docs.saltstack.com/en/develop/ref/runners/all/salt.runners.network.html

Reply:
Well, first of all, because I was not aware that it existed. Thanks for the reference.

Now that I read up on it, this built-in option has two advantages:
- it seems the runner can only wake machines which are on the same network segment as the master.  This option can use a distant minion to command the wakeup. In my use case, the hardware machine will often be in a different data center from the master.
- It does not require a separate operation. The data for this option is contained right in the profile where it is easy to keep track of.